Code/GraphMol/ChemReactions/ReactionParser.h

```diff
@@ -79,7 +79,7 @@ ChemicalReaction *RxnDataStreamToChemicalReaction(std::istream &rxnStream,
 ChemicalReaction *RxnSmartsToChemicalReaction(
     const std::string &text,
     std::map<std::string, std::string> *replacements = 0,
-    bool useSmiles = false);
+    bool useSmiles = false, bool sanitize=false);
```

Comment:
I think I would rather have a SanitizeRXN function as opposed to adding this to just the RxnSmarts code.  We have to sanitize .rxn files as well.

My current python implementation:

```
  1 fixes aromaticity
  2 merges queryhs if possible
  3 fixes up atom mappings and rgroup mappings from different sources
```

Perhaps it could be adopted to fix your issues as well.